api-test/DESIGN.md

```diff
@@ -1,82 +1,160 @@
 # Design of the Finesse Test Utility
 
+## Tools available
+
+There are tools that can integrate with Python or a script to accurately
+calculate API statistics. Currently, the needs are to test the tool using JSON
+files containing questions and their page origin in order to establish an
+accuracy score. We also want to calculate request times and generate a
+statistical summary of all this data. That being said, we plan to test the APIs
+under different conditions in the near future. For example, with multiple
+simultaneous users or under special conditions. That's why it's worth
+researching tools, if they are scalable and well adapted with Python.
+
+### Decision
+
+The most promising tool we found was [**Locust**](https://github.com/locustio).
+It seamlessly integrates with Python, making it a natural choice due to its
+dependency availability. It offers several advantages, as outlined below.
+However, **we decided to not use it** as its primary use case is to conduct
+tests with identical repeatable requests involving simultaneous users or
```

Comment:
Don't believe this is correct since arbitrary Python code can be provided to locust. 

So this Python code can sequentially load the JSON file and execute the post request using that data.

Reply:
There's a concept of plugin:

https://github.com/SvenskaSpel/locust-plugins

one such example of plugin is a csv reader:

https://github.com/SvenskaSpel/locust-plugins/blob/master/locust_plugins/csvreader.py

that reads from CSV:

https://github.com/SvenskaSpel/locust-plugins/blob/master/examples/csvreader_ex.py

we could come up with a jsonreader that reads from the directory where the json files are stored.

Reply:
I will work on a locust design and keep you up to date